crates/test-files/fixtures/demos/simple_open_auction.fe

```diff
@@ -0,0 +1,61 @@
+use std::context::Context
+
+contract SimpleOpenAuction:
+    # states
+    auction_end_time: u256
+    beneficiary: address
+
+    highest_bidder: address
+    highest_bid: u256
+
+    pending_returns: Map<address, u256>
+
+    ended: bool 
+
+    # events
+    event HighestBidIncreased:
+        idx bidder: address
+        amount: u256
+    
+    event AuctionEnded:
+        idx winner: address
+        amount: u256
+
+    
+    # constructor 
+    pub fn __init__(self, ctx: Context, bidding_time: u256, beneficiary_addr: address): 
+        self.beneficiary = beneficiary_addr
+        self.auction_end_time = ctx.block_timestamp() + bidding_time
+    
+    #method
+    pub fn bid(self, ctx: Context):
+        assert ctx.block_timestamp() <= self.auction_end_time, "Auction already ended"
```

Comment:
What do you think about encouraging the use of custom errors here.

E.g.

```
struct AlreadyEnded:
    pass
```

and then:

```
ctx.block_timestamp() > self.auction_end_time:
    revert AlreadyEnded()
```